src/codegen/templates/scale.py.mako

```diff
@@ -24,12 +25,13 @@ class Scale:
     """
     __slots__ = ['_name', '__weakref__']
 
-    def __init__(self, name):
+    def __init__(self, name,*, grpc_options=None, interpreter=None):
```

Comment:
`interpreter` is a private implementation detail. Instead of adding it to the public interface, let's use a simplified version of the `_TaskAlternateConstructor` approach.

```
class Scale:
    def __init__(self, name, *, grpc_options=None):
        self._name = name
        self._interpreter = utils._select_interpreter(grpc_options)
```

In a private submodule:
```
class _ScaleAlternateConstructor(Scale):
    def __init__(self, name, interpreter):
        self._name = name
        self._interpreter = interpreter
        self.__class__ = Scale
```

Reply:
Defining `_ScaleAlternateConstructor` in the same submodule as `Scale` sounds good. Shall we proceed on same?